yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -540,6 +565,28 @@ private[spark] class Client(
     amContainer
   }
 
+  def setupCredentials(): Unit = {
+    if (args.principal != null) {
+      if (args.keytab == null) {
```

Comment:
You could use `Preconditions.checkState()` (or `checkArgument`) for a shorter version.